omnibus/resources/agent/msi/cal/strings.cpp

```diff
@@ -118,6 +123,101 @@ void loadStringToWstring(int id, std::wstring *target)
 }
 static bool initialized = false;
 
+
+
+bool wrapGetComputerNameExW(COMPUTER_NAME_FORMAT fmt, std::wstring& result)
+{
+    wchar_t * buffer = NULL;
+    DWORD sz = 0;
+    BOOL res = GetComputerNameExW(fmt, buffer, &sz);
+    if(res) {
+        // this should never succeed
+        WcaLog(LOGMSG_STANDARD, "Unexpected.  Didn't get buffer size for computer name %d", (int)fmt);
+        return false;
+    }
+    DWORD err = GetLastError();
+    if(ERROR_MORE_DATA != err){
+        WcaLog(LOGMSG_STANDARD, "Unable to get computername info %d", err);
+        return false;
+    }
+    buffer = new wchar_t[sz + 1];
+    sz = sz + 1;
+    res = GetComputerNameExW(fmt, buffer, &sz);
+    if(res){
+        result = _wcslwr(buffer);
+    } else {
+        err = GetLastError();
+        WcaLog(LOGMSG_STANDARD, "Unable to get computername info %d", err);
+    }
+    if(buffer){
+        delete [] buffer;
+    }
+    return res;
+}
+void getHostInformation() {
+    wchar_t buf[MAX_COMPUTERNAME_LENGTH + 1];
+    DWORD sz = MAX_COMPUTERNAME_LENGTH+1;
+    if(!GetComputerNameW(buf, &sz)){
+        WcaLog(LOGMSG_STANDARD, "Failed to get computername %d", GetLastError());
+    } else {
+        computername = _wcslwr(buf);
+        WcaLog(LOGMSG_STANDARD, "Computername is %S (%d)", computername.c_str(), sz);
+    }
+    // get the computername again and compare, just to make sure
+    std::wstring compare_computer;
+    if(wrapGetComputerNameExW(ComputerNameDnsHostname, compare_computer)) {
+        if(computername != compare_computer){
+            WcaLog(LOGMSG_STANDARD, "Got two different computer names %S %S", computername.c_str(), compare_computer.c_str());
+        }
+    }
+    std::wstring domain; 
+    if(wrapGetComputerNameExW(ComputerNameDnsDomain, domain))
+    {
+        // newer domains will look like DNS domains.  (i.e. domain.local)
+        // just take the domain portion, which is all we're interested in.
+        size_t pos = domain.find(L'.');
```

Comment:
Could the domain had a `.` for some other reason? It could give problems then.

Reply:
No, `.` is a separator in a domain name.